src/internet/model/tcp-socket-base.cc

```diff
@@ -2035,59 +2044,84 @@ TcpSocketBase::ProcessSynSent (Ptr<Packet> packet, const TcpHeader& tcpHeader)
         }
       else
         {
           m_tcb->m_ecnState = TcpSocketState::ECN_DISABLED;
+          SendEmptyPacket ((TcpHeader::SYN | TcpHeader::ACK),false);
         }
-      SendPendingData (m_connected);
-      Simulator::ScheduleNow (&TcpSocketBase::ConnectionSucceeded, this);
-      // Always respond to first data packet to speed up the connection.
-      // Remove to get the behaviour of old NS-3 code.
-      m_delAckCount = m_delAckMaxCount;
     }
+  else if(tcpflags & (TcpHeader::SYN | TcpHeader::ACK)  
+         && m_tcb->m_nextTxSequence + SequenceNumber32 (1) == tcpHeader.GetAckNumber ()) 
+       {
+         
+        if(m_tcb->m_ecnState == TcpSocketState::ECN_CE_RCVD)
+         { 
+          if((m_ecnMode == EcnMode_t::EcnpTry) && (tcpflags & (TcpHeader::ECE))) // Received SYN+ACK with CE. Set the ECN state of the receiver to sending_ECE and send an ACK+ECE Packet.
+           {
+            NS_LOG_DEBUG ("Receiving SYN/ACK with CE mark");
+            m_retxEvent.Cancel ();
+            m_rxBuffer->SetNextRxSequence (tcpHeader.GetSequenceNumber () + SequenceNumber32 (1));
+            SendEmptyPacket ((TcpHeader::ACK | TcpHeader::ECE),false);
+            NS_LOG_DEBUG (TcpSocketState::EcnStateName[m_tcb->m_ecnState] << " -> ECN_SENDING_ECE");
+            m_tcb->m_ecnState = TcpSocketState::ECN_SENDING_ECE;
+           }
+         } 
+        else 
+         { 
+           /* Check if we received an ECN SYN-ACK packet. Change the ECN state of sender to ECN_IDLE if receiver has sent an ECN SYN-ACK
+           * packet and the  traffic is ECN Capable
+           */
+          if ((m_ecnMode == EcnMode_t::ClassicEcn || m_ecnMode == EcnMode_t::EcnpTry) && (tcpflags & (TcpHeader::ECE)))
+           {
+            NS_LOG_INFO ("Received ECN SYN-ACK packet.");
+            NS_LOG_DEBUG (TcpSocketState::EcnStateName[m_tcb->m_ecnState] << " -> ECN_IDLE");
+            m_tcb->m_ecnState = TcpSocketState::ECN_IDLE;
+           }
+          else
+           {
+            m_tcb->m_ecnState = TcpSocketState::ECN_DISABLED;
+           }
+          // Handshake completed
+          NS_LOG_DEBUG ("SYN_SENT -> ESTABLISHED");
+          m_congestionControl->CongestionStateSet (m_tcb, TcpSocketState::CA_OPEN);
+          m_state = ESTABLISHED;
+          m_connected = true;
+          m_retxEvent.Cancel ();
+          m_rxBuffer->SetNextRxSequence (tcpHeader.GetSequenceNumber () + SequenceNumber32 (1));
+          m_tcb->m_highTxMark = ++m_tcb->m_nextTxSequence;
+          m_txBuffer->SetHeadSequence (m_tcb->m_nextTxSequence);
+          SendEmptyPacket ((TcpHeader::ACK),false);
+          SendPendingData (m_connected);
+          Simulator::ScheduleNow (&TcpSocketBase::ConnectionSucceeded, this);
+          // Always respond to first data packet to speed up the connection.
+          // Remove to get the behaviour of old NS-3 code.
+          m_delAckCount = m_delAckMaxCount;
+         }
+      }
+
```

Comment:
Correct the spaces in the above chunk of code. A new line in a block starts after a space below the braces.